core/src/main/java/com/linecorp/armeria/client/endpoint/GradualStartEndpointSelectionStrategyBuilder.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.endpoint;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * A builder for creating a new {@link GradualStartEndpointSelectionStrategy} that ramps up the weight
+ * gradually.
+ */
+public final class GradualStartEndpointSelectionStrategyBuilder {
+    private EndpointWeightTransition endpointWeightTransition;
+    private ScheduledExecutorService executorService;
+    private Duration totalSlowStartDuration;
+    private Duration slowStartInterval;
+    private int numberOfSteps;
```

Comment:
Could we set some defaults here and simplify `GradualStartEndpointSelectionStrategy.of()`?